tests/meshcop/commissioner.hpp

```diff
@@ -0,0 +1,309 @@
+ */
+struct Context
+{
+    /** Set to true if we should commission the device */
+    bool commission_device;
+
+    /** coap instance to talk to agent & device */
+    Coap::Agent *mCoap;
+
+    /** dtls info for talking to agent & joiner */
+    Dtls::Server *mDtlsServer;
+
+    /* the session info for agent & joiner */
+    Dtls::Session *mSession;
+
+    /* dtls context information */
+    mbedtls_ssl_context *mSsl;
+    mbedtls_net_context *mNet;
+
+    /* Socket we are using with the agent */
+    int mSocket;
+
+    /* this generates our coap tokens */
+    uint16_t mCoapToken;
+
+    /* this is the commissioner session id */
+    uint16_t mCommissionerSessionId;
+
+    /** all things for the border agent */
+    struct br_agent
+    {
+
+        /** network port, in form mbed likes it (ascii) */
+        char mPort_ascii[ 7 ];
+
+        /** network address, in form mbed likes it (ascii) */
+        char mAddress_ascii[64];
+
+        /** xpanid from command line & binary form */
+        /* Note: xpanid is used to calculate the PSKc */
+        struct
+        {
+            char    ascii[  (kXpanidLength * 2) + 1 ];
+            uint8_t bin  [   kXpanidLength ];
+        } mXpanid;
+
+        /** network name from command line */
+        /* Note: networkname is used to calculate the PSKc */
+        char mNetworkName[ kNetworkNameLenMax + 1 ];
+
+        /** border router agent pass phrase */
+        /* Note: passphrase is used to calculate the PSKc */
+        char mPassPhrase[ kBorderRouterPassPhraseLen + 1 ];
+
+
+        /** the PSKc used with the agent */
+        struct
+        {
+
+            /* this class does the calculation */
+            Psk::Pskc mTool;
+
+            /** ascii & binary of PSKc, either from calculation or cmdline */
+            char    ascii[ (OT_PSKC_LENGTH * 2) + 1 ];
+            uint8_t bin[ OT_PSKC_LENGTH ];
+        } mPSKc;
+
+    } mAgent;
+
+    /** Kek with the joiner operation */
+    uint8_t mKek[32];
+
+    struct comm_ka
+    {
+        /** last time a COMM_KA message was sent */
+        struct timeval mLastTxTv;
+
+        /** last time a COMM_KA.rsp was received */
+        struct timeval mLastRxTv;
+
+        /** how often should these be sent in seconds */
+        int mTxRate;
+
+        /** How many have been sent & recieved */
+        int mTxCnt;
+        int mRxCnt;
+
+        /** are these disabled? (for test purposes) */
+        bool mDisabled;
+    } mCOMM_KA;
+
+    /** Total timeout (envelope) of the commissioning test */
+    struct               timeval mEnvelopeStartTv;
+    int                          mEnvelopeTimeout;
+
+    /** Commissioner state */
+    int mState;
+
+
+    /** All things about the joiner */
+    struct joiner
+    {
+
+        /** the EUI64 of the Joiner or HASHMAC, either from cmdline or computed */
+        struct
+        {
+            char    ascii[ (kEui64Len * 2) + 1 ];
+            uint8_t bin[ kEui64Len ];
+        } mEui64, mHashMac;
```

Comment:
same concern as `xpanid`, would it be better to handle parsing its argument handler?